ivy.el

```diff
@@ -3066,6 +3066,20 @@ RE-STR is the regexp, CANDS are the current candidates."
              (not (and ivy--flx-featurep
                        (eq ivy--regex-function 'ivy--regex-fuzzy)
                        (< (length cands) 200)))
+             ;; If there was a preselected candidate, don't try to
+             ;; keep it selected even if the regexp still matches
+             ;; it. See issue #1563. See also `ivy--preselect-index',
+             ;; which this logic roughly mirrors.
+             (not (or
+                   (and (integerp (ivy-state-preselect ivy-last))
+                        (= ivy--index
+                           (ivy-state-preselect ivy-last)))
+                   (equal (ivy-state-current ivy-last)
+                          (ivy-state-preselect ivy-last))
+                   (and (stringp (ivy-state-preselect ivy-last))
+                        (string-match-p
+                         (ivy-state-preselect ivy-last)
+                         (ivy-state-current ivy-last)))))
```

Comment:
Nitpick: might be nice to bind `ivy-state-preselect` and `ivy-state-current` to reduce visual clutter (and avoid duplicate calls).

Reply:
(Seeing as there are other calls to these accessors in the function as well.)